pkg/pool-weighted/contracts/managed/vendor/BasePool.sol

```diff
@@ -272,30 +239,22 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
             _mintPoolTokens(address(0), _getMinimumBpt());
             _mintPoolTokens(recipient, bptAmountOut - _getMinimumBpt());
 
-            // amountsIn are amounts entering the Pool, so we round up.
-            _downscaleUpArray(amountsIn, scalingFactors);
-
             return (amountsIn, new uint256[](balances.length));
         } else {
-            _upscaleArray(balances, scalingFactors);
             (uint256 bptAmountOut, uint256[] memory amountsIn) = _onJoinPool(
                 poolId,
                 sender,
                 recipient,
                 balances,
                 lastChangeBlock,
                 inRecoveryMode() ? 0 : protocolSwapFeePercentage, // Protocol fees are disabled while in recovery mode
```

Comment:
We can change the internal handler to not receive data we know we don't care about, such as the fee percentage, recipient, or last change block